Sources/Event Logging (Swift)/TrackEventPersistenceService.swift

```diff
@@ -0,0 +1,82 @@
+#if SWIFT_PACKAGE
+import AutomatticTracksModel
+import AutomatticTracksModelObjC
+#endif
+import CoreData
+
+/// Should eventually replace objC class `TracksEventPersistenceService`.  Unfortunately
+/// Swift Packages do not support mixed source classes so we'll instead need to compose.
+///
+@objc
+public class TrackEventPersistenceService: NSObject {
+    
+    private let managedObjectContext: NSManagedObjectContext
+    
+    @objc
+    public init(managedObjectContext: NSManagedObjectContext) {
+        self.managedObjectContext = managedObjectContext
+        
+        super.init()
+    }
+    
+    /// Increments the retry count for the specified events in batches of
+    @objc
+    public func incrementRetryCountForEvents(_ tracksEvents: [TracksEvent], onComplete completion: ((Error?) -> Void)?) {
+        let uuidStrings = tracksEvents.map { event in
+            event.uuid.uuidString
+        }
+        
+        let batchSize = 500
+        
+        for startIndex in stride(from: 0, to: uuidStrings.count, by: batchSize) {
+            let isLastBatch = startIndex + batchSize >= uuidStrings.count
+            
+            managedObjectContext.perform {
+                let results: [TracksEventCoreData]
+                let count = min(uuidStrings.count - startIndex, batchSize)
+                let uuidStringsBatch = Array(uuidStrings[startIndex ..< startIndex + count])
+                
+                do {
+                    results = try self.findTrackEventCoreData(uuidStrings: uuidStringsBatch)
+                } catch {
+                    TracksLogError("Error while finding track events: \(String(describing: error))")
+                    
+                    if isLastBatch {
+                        completion?(error)
```

Comment:
This is kinda awful in that we'll only see an error if the last call to `findTrackEventCoreData` fails.

IMHO there are two options here:

1. Ignore the partial errors and not return any error ever... just log them.
2. Create a type of error that groups all partial errors and return that instead... although that error could still mean the retry count was successfully incremented for some events.  I'm more prone to rely on option 1 because I don't think these errors are relevant to whoever's calling this method.

Reply:
The previous code had no room for reporting an error either; maybe we don't even need the `error` parameter in the completion handler here at all? I don't think the caller is ever likely to actually do something about an error here.